test.sh

```diff
@@ -84,7 +132,29 @@ for test_name in "${TEST_NAMES[@]}"; do
     if ! time ct install "${CT_ARGS[@]}" --helm-extra-set-args "$HELM_EXTRA_SET_ARGS ${testCases[$test_name]}"; then
         echo 1>&2 "❌ Test $test_name failed"
         echo 1>&2 "Test logs:"
-        kubectl --namespace "$NAMESPACE" logs --tail=-1 --selector app.kubernetes.io/component=test
+        # Get list of failed test pods
+        failed_tests=$(kubectl get pods --namespace "$NAMESPACE" \
+                    --field-selector=status.phase=Failed \
+                    --selector app.kubernetes.io/component=test \
+                    -o jsonpath='{.items[*].metadata.name}')
+
+        # Iterate through each failed test pod
+        for pod in $failed_tests; do
+            # Get list of failed containers in the pod
+            containers=$(kubectl get pod "$pod" --namespace "$NAMESPACE" \
+                -o jsonpath="{range .status.containerStatuses[?(@.state.terminated.exitCode != 0)]}{.name}{' '}{end}")
+
+            # Check if there are any failed containers
+            if [ -n "$containers" ]; then
+                # Iterate through each failed container
+                for container in $containers; do
```

Comment:
This is getting complex. Shellcheck will complain that `$containers` is not quoted, and it should be an array. Maybe we can just print all container logs with `kubectl logs --all-containers=true`, and make sure those who are successful don't print unnecessary output?

Reply:
This option completely makes sense to me. I just realized `test-connection.yaml` currently logs a fair bit of stuff even on success.

Reply:
You can remove `--network-logging=BODY` from it, I was too eager to add it (never needed it). Maybe add `--no-progress`, I'm not sure if it automatically detects there's no TTY, but explicit is better than implicit anyway.